nemo/collections/asr/parts/utils/streaming_utils.py

```diff
@@ -1371,11 +1379,18 @@ def __iter__(self):
             # if there is not enough frames to be used as the pre-encoding cache, zeros would be added
             zeros_pads = None
             if self.buffer_idx == 0 and isinstance(self.streaming_cfg.pre_encode_cache_size, list):
-                cache_pre_encode = torch.zeros(
-                    (audio_chunk.size(0), self.input_features, self.streaming_cfg.pre_encode_cache_size[0]),
-                    device=audio_chunk.device,
-                    dtype=audio_chunk.dtype,
-                )
+                if self.pad_and_drop_preencoded:
```

Comment:
How about saving the value of self.streaming_cfg.pre_encode_cache_size[X] in  a variable and then use it to reduce the length of duplication?